Concurrency.playground/Contents.swift

```diff
@@ -0,0 +1,86 @@
+import UIKit
+
+
+class Spoon {
+    
+    private var lock = NSLock()
+    var index: Int
+   
+    func pickUp() {
+        lock.lock()
+    }
+    
+    func putDown() {
+        lock.unlock()
+    }
+    
+    
+    init(index: Int) {
+        self.index = index
+    }
+    
+}//
+
+
+class Developer {
+   
+    var leftSpoon: Spoon
+    var rightSpoon: Spoon
+    
+    
```

Comment:
Would be nice to add a name or id or something, so that you can use print statements to see what it happening.